### ISSUE TITLE
Always getting NaNs in long training

### ISSUE THREAD

**Issue by danbochman** [lucidrains/lion-pytorch#33]
I've tried using the LION optimizer in your other (great) Imagen repository. I can share my anecdotal experience:

- I was experimenting with LION with models of different sizes (depth and width)
- I experimented with other betas such `beta1 0.95` and `beta2 0.98`
- I experimented with learning rates 1e-4, 3e-5 and 1e-5.
- I experimented with the triton kernel turned both on and off.

Training was indeed fast but unfortunately in the end always ended up yielding NaNs.

I think a potential issue could be how LION interacts with a warmup schedule, I am not sure if you're supposed to do warmup with this optimizer or not (which I always did). 

![image](https://github.com/lucidrains/lion-pytorch/assets/40274004/7159adce-ecdd-48c6-909f-116a4a948e23)

**Comment by ysesst93013** [comment]
I have same problem :(

**Comment by SergeySakharovskiy** [comment]
same NaN issue with CosineAnnealing scheduler after the first epoch.

**Comment by xiangning-chen** [comment]
> I've been experimenting with the LION optimizer in your other (great) Imagen repository. I can share my anecdotal experience and combinations:
> 
> * Models of different sizes 0.2B, 0.7B and 1B params.
> * Betas such as `beta1 0.95` and `beta2 0.98`
> * Learning rates `1e-4`, `3e-5` and `1e-5`.
> * Triton kernel turned both `True` and `False`.
> 
> Training was indeed fast but unfortunately in the end always ended up yielding NaNs.
> 
> I think a potential issue could be how LION interacts with a warmup schedule; I am not sure if you're supposed to do warmup with this optimizer or not (which I always did).
> 
> ![image](https://private-user-images.githubusercontent.com/40274004/288020853-7159adce-ecdd-48c6-909f-116a4a948e23.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY1NjcwNTAsIm5iZiI6MTcwNjU2Njc1MCwicGF0aCI6Ii80MDI3NDAwNC8yODgwMjA4NTMtNzE1OWFkY2UtZWNkZC00OGM2LTkwOWYtMTE2YTRhOTQ4ZTIzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI5VDIyMTkxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUzYjgyYWM3NjBhYTFjMzE5N2Q1MDNhNmIwZTdkOGJjOWRhNDQ2YmQ5ZWQ2ODU0NmZkNDVkOGQwYzhjYzFhMjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OhD5opPldpyYJ5iBQTQ121f3f93aMqlGkQ1CJnDMeMI)

May I know the learning rate schedule you are using?

**Comment by zjutzyl** [comment]
same issue,  i set a big weight decay to avoid it.  i suppose that  'update=symbol * lr' enlarging abs(parameter) while symbol not changing.

**Comment by lindakasabian** [comment]
same here. sudden nan losses during 100 e training with onecyclelr
